universal-login-sdk/test/counterfactual.ts

```diff
@@ -20,10 +21,26 @@ describe('SDK counterfactual', () => {
     sdk = new UniversaLoginSDK(relayer.url(), provider);
   });
 
-  it('getFutureWallet returns private key and contract address', async () => {
-    const [privateKey, futureContractAddress] = (await sdk.getFutureWallet(() => {}, () => {}));
+  it('createFutureWallet returns private key and contract address', async () => {
+    const {privateKey, contractAddress} = await sdk.createFutureWallet();
     expect(privateKey).to.be.properPrivateKey;
-    expect(futureContractAddress).to.be.properAddress;
+    expect(contractAddress).to.be.properAddress;
+  });
+
+  it('waitForBalance returns promise, which resolves when balance update', async () => {
+    const {waitForBalance, contractAddress} = (await sdk.createFutureWallet());
+    setTimeout(() => wallet.sendTransaction({to: contractAddress, value: utils.parseEther('2')}), 50);
+    const result = await waitForBalance();
+    expect(result.contractAddress).be.eq(contractAddress);
+    expect(result.tokenAddress).be.eq(ETHER_NATIVE_TOKEN.address);
+  });
+
+  it('should deploy contract with future address', async () => {
+    const {deploy, contractAddress, waitForBalance} = (await sdk.createFutureWallet());
+    await wallet.sendTransaction({to: contractAddress, value: utils.parseEther('2')});
+    await waitForBalance();
+    const deployedContractAddress = await deploy('name.mylogin.eth');
+    expect(deployedContractAddress).to.be.eq(contractAddress);
```

Comment:
Not good enough expect - check for the code under the address.
Like here: https://github.com/UniversalLogin/UniversalLoginSDK/blob/1124ee994c7cd5a833e518f5f85cadfb57af9daa/universal-login-sdk/lib/services/BlockchainService.ts#L8